submissions/Dovahkiin1991/oop-exercise/index.js

```diff
@@ -0,0 +1,76 @@
+import { print } from './js/lib.js';
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Dovahkiin1991/a-tiny-JS-world
+   Web app: https://dovahkiin1991.github.io/a-tiny-JS-world/
+   */
+// ======== OUTPUT ========
+const UNSET_VALUE = 'unset';
+class Creature {
+    constructor(species, name, gender, legs, hands, saying) {
+        this.species = species;
+        this.name = name;
+        this.gender = gender;
+        this.legs = legs;
+        this.hands = hands;
+        this.saying = saying;
+    }
+
+    listOfProperties() {
+        return `
+            species - ${this.species = this.species || UNSET_VALUE};
+            name - ${this.name = this.name || UNSET_VALUE}; 
+            gender - ${this.gender = this.gender || UNSET_VALUE}; 
+            legs - ${this.legs = this.legs || UNSET_VALUE}; 
+            hands - ${this.hands = this.hands || UNSET_VALUE}; 
```

Comment:
A class shouldn't hadle properties it doesn't own. This is the purpose of encapsulation.
OOP provides for mechanics that allow classes to manage their own properties.
Look into method overloading/overriding.